proposals/exception-handling/Exceptions.md

```diff
@@ -205,99 +193,166 @@ flushed, the embedder defines how to handle uncaught exceptions. Otherwise, the
 found enclosing try block is the catching try block.
 
 A throw inside the body of a catch block is never caught by the corresponding
-try block of the catch block, since instructions in the body of the catch block
-are not in the body of the try block.
+try block of the catch/unwind block, since instructions in the body of the
+catch/unwind block are not in the body of the try block.
 
 Once a catching try block is found for the thrown exception, the operand stack
-is popped back to the size the operand stack had when the try block was entered,
-and then an `exnref` referring to the caught exception is pushed back onto the
-operand stack.
+is popped back to the size the operand stack had when the try block was entered.
+
+Then in case of a try-catch block, tagged catch blocks are tried in the
+order they appear in the catching try block, until one matches. If a matched
+tagged catch block is found, control is transferred to the body of the catch
+block, and the data fields of the exception are pushed back onto the stack.
+
+Otherwise, control is transferred to the body of the `catch_all` block, if any.
+However, unlike tagged catch blocks, the constructor arguments are not copied
+back onto the operand stack. If no tagged catch blocks were matched, and the
+catching try block doesn't have a `catch_all` block, the exception is rethrown
+to the next enclosing try block.
 
 If control is transferred to the body of a catch block, and the last instruction
 in the body is executed, control then exits the try block.
 
-If the selected catch block does not throw an exception, it must yield the
-value(s) expected by the corresponding catching try block. This includes popping
-the caught exception.
+In case of a try-unwind block, the control is transfered to the body of the
+`unwind` block. After executing instructions within the `unwind` block, the
+exception is automatically rethrown when the control reaches the `end`
+instruction. As in the case of the `catch_all` block, the exception arguments
+are not copied onto the operand stack.
+
+If the selected catch/unwind block does not throw an exception, it must yield
+the value(s) expected by the corresponding catching try block. This includes
+popping the caught exception.
```

Comment:
It is also possible for a catch block to do nothing with the exception and have the correct type if the type of the exception matches the type of the try.

```suggestion
If the selected catch/unwind block does not throw an exception, it must yield
the value(s) expected by the corresponding catching try block.
```

Reply:
I agree that text is confusing. Removed.